tool/tsh/common/device.go

```diff
@@ -98,18 +111,36 @@ func (c *deviceEnrollCommand) run(cf *CLIConf) error {
 			return trace.Wrap(err)
 		}
 		defer authClient.Close()
-
 		devices := authClient.DevicesClient()
-		dev, err = enroll.RunCeremony(ctx, devices, cf.Debug, c.token)
+
+		enrollCeremony := enroll.NewCeremony()
+		if c.currentDevice {
+			dev, outcome, err = enrollCeremony.RunAdmin(ctx, devices, cf.Debug)
+			// err handled below.
+		} else {
+			dev, err = enrollCeremony.Run(ctx, devices, cf.Debug, c.token)
+			if err == nil {
+				outcome = enroll.DeviceEnrolled
+			}
+			// err handled below.
+		}
 		return trace.Wrap(err)
 	}); err != nil {
 		return trace.Wrap(err)
 	}
 
+	var action string
+	switch outcome {
+	case enroll.DeviceRegisteredAndEnrolled:
```

Comment:
Consider implementing `fmt.Stringer` on this enum, so you can just include the outcome in your printf.

Reply:
+1

Reply:
I like the dumb, rather explicit switch block. It makes it clear that these are user-visible messages; I would expect a typical enum Stringer to print the enum name I read in code instead.